infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -906,13 +908,20 @@ private void setupLocalNodeId() {
             File nodeIdFile = new File(nodeIdFilePath);
             try (BufferedReader bufferedReader = new BufferedReader(new FileReader(nodeIdFile))) {
                 String line = bufferedReader.readLine();
-                localNodeId = Optional.of(line.split("=")[1].trim().toLowerCase());
+                localNodeId = line.split("=")[1].trim().toLowerCase();
```

Comment:
why remove Optional?

Reply:
Optional was introduced to use IP address when the node id is missing. Now LR should fail if the node id is missing because it is not correct to identify a node by its IP address, which is mutable.